nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -529,46 +549,58 @@ def _aggregate_impl(self):
         return self._container.shape
 
 
-class MinAggregator(TensorAggregatorBase):
-    def _register_reduced_input_impl(self, x: TensorType) -> None:
-        if not self._container:
-            self._container = x
+class OnlineOfflineAggregatorBase(TensorAggregatorBase, ABC):
+    def __init__(
+        self,
+        tensor_processor: NNCFCollectorTensorProcessor,
+        aggregation_axes: Optional[Tuple[int, ...]] = None,
+        num_samples: Optional[int] = None,
+        window_size=None,
+    ):
+        super().__init__(tensor_processor, aggregation_axes=aggregation_axes, num_samples=num_samples)
+        self._window_size = window_size
+        self._container = deque(maxlen=window_size)
+
+
+class OnlineAggregatorBase(OnlineOfflineAggregatorBase, ABC):
+    def _online_register_reduced_input_impl(self, x: TensorType, fn) -> None:
```

Comment:
I prefer functional style, and as I specify recurred function signature it will be explicit. What are disadvantages for this method?

Reply:
You should change your preferences, then. 

Specifying a type hint for a callable is less readable that a function signature of a method in a class. 

Passing a function implies that the callable could be dynamically (runtime) determined, which is not true in our case - in our case the static specification via proper class methods and inheritance is enough. 

You are unnecessarily inverting the way the action of a code line is understood - instead of passing varying parameters to a function call, which is how most of the code is written, you are passing a varying function and making the dev look up the way that the function is used in actual code in a completely another place. 

Look through the https://github.com/openvinotoolkit/nncf/pull/2120/files#diff-6e8ff64441fe5f0da52a90b8958f05b9525f20e57e9e85916a3d0193bb495f1d and see for yourself that the code for the `MeanNoOutliersAggregator`, for instance, has become easier to understand - it is immediately visible just how this class is different from others, and the code that does the actual difference is right there in the definition of the same class - you don't need to jump between the superclasses' `_offline_aggregation_impl` and reference it against the current value of `fn` anymore.

Reply:
Valid comment. As this code style is already spreaded across all aggregators and function inside `TensorCollectorProcessor` (like `no_outliers_map`, `NoOutliersAggregatorBase`), I suggest to put this request to a refactoring request list, what do you think?

Reply:
I've done the refactoring that you mean in the PR I linked, it should be trivial to incorporate the results into your PR here.

Reply:
Done, please check